convert2rhel/checks.py

```diff
@@ -95,56 +97,59 @@ def check_convert2rhel_latest():
-    latest_version = ("0", "0.00", "0")
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determin the latest_available_version in the convert2rhel yum repo
+    # it assigns the epoch ,version, and release ex:("0" "0.26" "1.el7") to the latest_available_version var.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.lableCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if lastest_version is greater then package_version, 0 if they are equal, 1 if package_version is greater the lastest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
-
-    # After the for loop latest_version will have the epoch ,version, and release ex:("0" "0.26" "1.el7") information from convert2rhel yum repo.
-    # the release for latest_verion will be "1.el7" and the relase for convert2rhel_version will be hard coded as "0" below,
-    # therefore when the versions are the same the latest_version's release field will cause it to evaluate as later
-    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
+            latest_available_version = package_version[1:]
+
+    # we hard code 0 for the epoch and release below because latest_verion will
+    # contain "1.el7" for the release but we don't know what the release for
+    # installed_convert2rhel_version will be, if we did not hard code to 0 latest_available_version
+    # will evaluate as later then installed_convert2rhel_version even when the versions
+    # were the same.
+    #
+    # ver_compare is set to
+    # * -1 if lastest_version is greater then installed_convert2rhel_version,
+    # * 0 if they are equal
+    # * 1 if installed_convert2rhel_version is greater the lastest_version
+    ver_compare = rpm.labelCompare(("0", installed_convert2rhel_version, "0"), ("0", latest_available_version[1], "0"))
     if ver_compare < 0:
         if "CONVERT2RHEL_UNSUPPORTED_VERSION" in os.environ:
             logger.warning(
                 "You are currently running %s and the latest version of convert2rhel is %s.\n"
-                "'CONVERT2RHEL_UNSUPPORTED_VERSION' environment detected, continuing conversion"
-                % (convert2rhel_version, latest_version[1])
+                "'CONVERT2RHEL_UNSUPPORTED_VERSION' environment variable detected, continuing conversion"
+                % (installed_convert2rhel_version, latest_available_version[1])
             )
 
         else:
             if int(system_info.version.major) <= 6:
```

Comment:
I think that you don't need to cast the `system_info.version.major` here to an integer, as this is intended to be an integer already. 

You can see other examples in the codebase that don't have this implicit cast to `int()`. Let me know if you had to add this because it was failing without the cast (it shouldn't).

Reply:
I'm not sure exactly why you are seeing this var like this. I don't see this in my code, but I do have this in my current code and have committed it!